rpc/src/v1/impls/signing_unsafe.rs

```diff
@@ -38,106 +37,101 @@ use v1::types::{
 };
 
 /// Implementation of functions that require signing when no trusted signer is used.
-pub struct SigningUnsafeClient<C, M> where
-	C: MiningBlockChainClient,
-	M: MinerService,
-{
+pub struct SigningUnsafeClient<D> {
 	accounts: Weak<AccountProvider>,
-	client: Weak<C>,
-	miner: Weak<M>,
+	dispatcher: D,
 }
 
-impl<C, M> SigningUnsafeClient<C, M> where
-	C: MiningBlockChainClient,
-	M: MinerService,
-{
-
+impl<D: Dispatcher + 'static> SigningUnsafeClient<D> {
 	/// Creates new SigningUnsafeClient.
-	pub fn new(client: &Arc<C>, accounts: &Arc<AccountProvider>, miner: &Arc<M>)
-		-> Self {
+	pub fn new(accounts: &Arc<AccountProvider>, dispatcher: D) -> Self {
 		SigningUnsafeClient {
-			client: Arc::downgrade(client),
-			miner: Arc::downgrade(miner),
 			accounts: Arc::downgrade(accounts),
+			dispatcher: dispatcher,
 		}
 	}
 
-	fn handle(&self, payload: RpcConfirmationPayload, account: DefaultAccount) -> Result<RpcConfirmationResponse, Error> {
-		let client = take_weak!(self.client);
-		let miner = take_weak!(self.miner);
-		let accounts = take_weak!(self.accounts);
+	fn handle(&self, payload: RpcConfirmationPayload, account: DefaultAccount) -> BoxFuture<RpcConfirmationResponse, Error> {
+		let setup = move || {
+			let accounts = take_weak!(self.accounts);
+			let default_account = account;
```

Comment:
Why is this line needed? Similar to `dispatch` could be greatly simplified with `take_weak` for futures